frontend/web/components/metadata/AddMetadataToEntity.tsx

```diff
@@ -0,0 +1,251 @@
+
+type CustomMetadataField = MetadataField & {
+  metadataModelFieldId: number | string | null
+  isRequiredFor: boolean
+  model_field?: string | number
+}
+
+type CustomMetadata = (Metadata & CustomMetadataField) | null
+
+type AddMetadataToEntityType = {
+  organisationId: string
+  projectId: string | number
+  entityContentType: number
+  entityId: string
+  entity: number | string
+  createMetadataField: () => void
+  updateMetadata: () => void
+  getMetadata: (m: CustomMetadataField[]) => void
+}
+
+const AddMetadataToEntity: FC<AddMetadataToEntityType> = ({
+  entity,
+  entityContentType,
+  entityId,
+  getMetadata,
+  organisationId,
+  projectId,
+}) => {
+  const { data: metadataFieldList, isSuccess: metadataFieldListLoaded } =
+    useGetMetadataFieldListQuery({
+      organisation: organisationId,
+    })
+  const {
+    data: metadataModelFieldList,
+    isSuccess: metadataModelFieldListLoaded,
+  } = useGetMetadataModelFieldListQuery({
+    organisation_id: organisationId,
+  })
+
+  const { data: projectFeatureData, isSuccess: projectFeatureDataLoaded } =
+    useGetProjectFlagQuery(
+      {
+        id: entityId,
+        project: projectId,
+      },
+      { skip: entity !== 'feature' || !entityId },
+    )
+
+  const { data: segmentData, isSuccess: segmentDataLoaded } =
+    useGetSegmentQuery(
+      {
+        id: `${entityId}`,
+        projectId: `${projectId}`,
+      },
+      { skip: entity !== 'segment' || !entityId },
+    )
+
+  const [
+    metadataFieldsAssociatedtoEntity,
+    setMetadataFieldsAssociatedtoEntity,
+  ] = useState<CustomMetadataField[]>()
+
+  useEffect(() => {
+    getMetadata?.(metadataFieldsAssociatedtoEntity!)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [metadataFieldsAssociatedtoEntity])
+
+  useEffect(() => {
+    if (
+      metadataFieldList &&
+      metadataFieldListLoaded &&
+      metadataModelFieldList &&
+      metadataModelFieldListLoaded
+    ) {
+      // Filter metadata fields based on the provided content type
+      const metadataForContentType = metadataFieldList.results
+        // Filter metadata fields that have corresponding entries in the metadata model field list
+        .filter((meta) => {
+          return metadataModelFieldList.results.some((item) => {
+            return (
+              item.field === meta.id && item.content_type === entityContentType
+            )
+          })
+        })
+        // Map each filtered metadata field to include additional information from the metadata model field list
+        .map((meta) => {
+          // Find the matching item in the metadata model field list
+          const matchingItem = metadataModelFieldList.results.find((item) => {
+            return (
+              item.field === meta.id && item.content_type === entityContentType
+            )
+          })
+          // Determine if isRequiredFor should be true or false based on is_required_for array
+          const isRequiredFor =
```

Comment:
can all be simplified to matchingItem?.is_required_for.length > 0

Reply:
Is this logic actually correct, It could be bad variable naming but is "is_required_for" a list of entity types it's required for? If so, should this be checking is_required_for contains "FEATURE" or something?

Reply:
simplified, and it is a list of the content types id of the entities.